dali/pipeline/executor/executor2/exec_node_task.cc

```diff
@@ -367,11 +369,15 @@ OpTask::OpTaskOutputs OpTask::GetWorkspaceOutputs() {
       // The consumer stream will be properly synchronized in SetWorkspaceInputs.
       // This is done to facilitate freeing of memory - if we're able to transfer the
       // object to the consumption stream, it'll be freed in consumption order.
-      if (!ptr->shares_data()) {
-        if (AccessOrder consumer_order = OutputConsumerStream(o))
-          ptr->set_order(consumer_order, false);
+      if (ptr->is_pinned()) {
```

Comment:
This fixes a bug - previously we would set the consumer order for non-pinned host buffers - this PR adds an assert that verifies that such buffers are always in host order.